app/controllers/api/v1/foods_controller.rb

```diff
@@ -0,0 +1,56 @@
+class Api::V1::FoodsController < ApplicationController
+  def index
+    render json: Food.all
+  end
+
+  def show
+    food = Food.where(id: params[:id]).first
+
+    render nothing: true, status: 404 and return if food.nil?
+    render json: food
+  end
+
+  def create
+    food = Food.new(food_params)
+
+    if food.save
+      render json: food
+    else
+      render nothing: true, status: 400
+    end
+  end
+
+  def update
+    food = Food.where(id: params[:id]).first
+
+    if food.nil?
+      render nothing: true, status: 400 and return
+    end
+
+    if food.update(food_params)
+      render json: food
+    else
+      render nothing: true, status: 400
+    end
+  end
+
+  def destroy
+    food = Food.where(id: params[:id]).first
+
+    if food.nil?
+      render nothing: true, status: 404 and return
+    end
+
+    if food.delete
+      render nothing: true, status: 204
+    else
+      render nothing: true, status: 404 and return
```

Comment:
@corywest I feel like this is possibly redundant? As thought it is one of those "it could happen, however unlikely" errors. Not sure how to properly test this behavior, as it would have to fail at the SQL transaction point presumably (as if it was some error with the database) or am I wrong about that? Is there a decent way to test something like this? Or for the scope of this project, should I not bother digging too deep into an edge case like that.

Nearly-identical situation on like 33

Reply:
@kolyaventuri Yes, it's a bit redundant. You could assume that a SQL transaction itself could fail at some point, but it's unlikely. I would throw that test case out for now. We can chat more about it later if you'd like!

Reply:
Excellent, thanks! I would definitely like to talk about it later, how to handle/test it and such